src/urllib3/connectionpool.py

```diff
@@ -547,7 +547,7 @@ def urlopen(  # type: ignore[override]
         pool_timeout: Optional[int] = None,
         release_conn: Optional[bool] = None,
         chunked: bool = False,
-        body_pos: Optional[Union[int, object]] = None,
+        body_pos: _TYPE_FAILEDTELL = None,
```

Comment:
I _think_ we may want to keep the `Union` here as well. If we're going to define a variable for the composed type, the name probably needs to be more specific to `body_pos` rather than _FAILEDTELL. That's a side effect that we should try to minimize in the public typing. It's never intended for a user to pass this.

Reply:
I read through the other timeout PR and think I have a better grasp on what was done there. I'll enumerate the concepts we want to have defined with this change.

`_FAILEDTELL`: This is a sentinel value we created to denote we'd tried to get the position of our body object by calling `tell()` but it failed for whatever reason. It should never be passed as an argument by the user.

`_TYPE_FAILEDTELL`: This is the type of the sentinel object itself (_FAILEDTELL). I'm not sure if we want to conflate defaults with this concept. It may be more accurate to use either `_TYPE_SENTINEL` or `_TYPE_FAILEDTELL` in place of `_TYPE_DEFAULT`.

For how we handle this with `body_pos`, I think the type should be something closer to `Optional[Union[int, _TYPE_FAILEDTELL]]`. We can either use this as a standalone value or name it something like `_TYPE_BODY_POSITION`. I don't think we want to type any public interfaces with `_TYPE_FAILEDTELL` to avoid confusion.

Reply:
yes, this makes sense will rename based on this.